app/src/main/kotlin/compiler/intermediate_form/ControlFlowGraphBuilder.kt

```diff
@@ -9,8 +9,15 @@ class ControlFlowGraphBuilder(@JvmField var entryTreeRoot: IFTNode? = null) {
     private var conditionalTrueLinks = referenceHashMapOf<IFTNode, IFTNode>()
     private var conditionalFalseLinks = referenceHashMapOf<IFTNode, IFTNode>()
     private var treeRoots = ArrayList<IFTNode>()
-    private val finalTreeRoots: List<IFTNode> get() = treeRoots.filter {
-        it !in unconditionalLinks && it !in conditionalTrueLinks && it !in conditionalFalseLinks
+
+    val finalTreeRoots: List<Pair<IFTNode, CFGLinkType>> get() = treeRoots.filter {
+        it !in unconditionalLinks && (it !in conditionalTrueLinks || it !in conditionalFalseLinks)
+    }.map {
+        it to when (it) {
+            in conditionalTrueLinks -> CFGLinkType.CONDITIONAL_FALSE
+            in conditionalFalseLinks -> CFGLinkType.CONDITIONAL_TRUE
+            else -> CFGLinkType.UNCONDITIONAL
+        }
```

Comment:
```suggestion
    val finalTreeRoots: List<Pair<IFTNode, CFGLinkType>> = build().finalTreeRoots
```
?

Reply:
Actually, in some places in this file it is used as `build().finalTreeRoots`, is there a difference?

Reply:
Idk, I haven't originally written this code. This property was copy-pasted from `ControlFlowGraph` and so I changed the copy.